meson.build

```diff
@@ -0,0 +1,229 @@
+project('libass', 'c', license: 'ISC',
+        meson_version: '>= 0.49.0',
+        default_options: ['c_std=gnu99'],
+        version: '0.14.0')
+
+conf = configuration_data()
+deps = []
+
+if host_machine.system() == 'darwin'
+    add_languages('objc')
+endif
+
+if get_option('large-tiles')
+    conf.set('CONFIG_LARGE_TILES', 1)
+endif
+
+cc = meson.get_compiler('c')
+if cc.get_argument_syntax() == 'gcc'
+    cflags = [
+        '-Wall',
+        '-Wextra',
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-Werror-implicit-function-declaration', 
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+link_args = cc.get_supported_link_arguments(['-prefer-non-pic'])
+
+check_headers = [
+    'dlfcn.h',
+    'iconv.h',
+    'inttypes.h',
+    'memory.h',
+    'stdbool.h',
+    'stdint.h',
+    'stdlib.h',
+    'string.h',
+    'strings.h',
+    'sys/stat.h',
+    'sys/types.h',
+    'unistd.h'
+]
+
+check_functions = [
+    'strdup',
+    'strndup'
+]
+
+m_dep = cc.find_library('m', required: false)
+if m_dep.found()
+    deps += m_dep
+    conf.set('HAVE_LIBM', 1)
+endif
+
+iconv_dep = cc.find_library('iconv', required: false)
+if iconv_dep.found()
+    deps += iconv_dep
+    conf.set('CONFIG_ICONV', 1)
+endif
```

Comment:
Does this also deal with cases where iconv_open is a part of the system's libc *(eg glibc)* and thus **no** library is required for it to be available?

Reply:
Probably not? Is it sufficient to just check for a random iconv function to cover that case, `cc.has_function('iconv_open')` or something?

Reply:
Checking for `iconv_open` to be available at compile time should be sufficient. *(During link time some implementations will instead use `libiconv_open`)*